### DISCUSSION TITLE
Read and write bnd_s2 instances

### DISCUSSION THREAD

**Discussion by rchen234** [ambros-gleixner/MIPcc23#20] [Q&A] [answered]
Hi,
I tried to write bnd_s2 instances as lp files. Apparently the writeLP() function in the current version of pyscipopt does not work properly. Then I tried to convert the instance from mps format to lp format using gurobipy. When reading the instance, the following warnings popped up:
"Warning: duplicate bounds for column x2.
Warning: duplicate bounds for column x6.
..."
And it seems like Guoribi is completely ignoring the additional variable bounds on these variables x2, x6 etc. (The log files look identical to each other after calling the optimize function for all bnd_s2 instances when using Gurobi. This does to happen to SCIP.)

I wonder if it is a Gurobi bug and how we can properly human-read such instances.

Thanks!

**Comment by sbolusani** [comment, marked as answer]
Dear Rui,

You can use the `writeProblem()` method of PySCIPOpt to convert MPS format files into LP format files (example: `model.writeProblem("file_name.lp")`; note that the extension "lp" is necessary). The `writeLP()` method writes the current node's linear relaxation to a file (which is not needed here).

Best,
Suresh

**Reply by rchen234** [reply]
Thank you, Suresh! It works!
